app/src/pages/Add_Review.py

```diff
@@ -0,0 +1,160 @@
+    position_data = {
+        "companyID": company_id,
+        "name": position_name,
+        "description": "New position",
+        "remote": remote
+    }
+    try:
+        # Generate a unique positionID
+        response = requests.get(f'http://api:4000/po/positions')
+        response.raise_for_status()
+        existing_positions = response.json()
+        existing_ids = {position['positionID'] for position in existing_positions}
+        new_position_id = max(existing_ids) + 1 if existing_ids else 1
+
+        position_data["positionID"] = new_position_id
+
+        response = requests.post(f'http://api:4000/po/positions', json=position_data)
+        response.raise_for_status()
+        return response.json()['positionID']
+    except requests.exceptions.RequestException as e:
+        st.error(f"Failed to create position: {e}")
+        return None
+
+def get_author_id_by_name(author_name):
+    try:
+        response = requests.get(f'http://api:4000/rver/reviewers?name={author_name}')
+        response.raise_for_status()
+        author_data = response.json()
+        if author_data:
+            return author_data[0]['reviewerID']
+        else:
+            st.error("Reviewer not found")
+            return None
+    except requests.exceptions.RequestException as e:
+        st.error(f"Failed to fetch reviewer data: {e}")
+        return None
+
+def add_review(review_data):
+    try:
+        response = requests.post('http://api:4000/r/reviews', json=review_data)
+        response.raise_for_status()
+        st.success("Review added successfully!")
+        st.experimental_rerun()
+    except requests.exceptions.RequestException as e:
+        st.error(f"Failed to add review: {e}")
+
+def main():
+    st.title("Add a Review")
+    
+    company_names = get_company_names()
+    if not company_names:
+        return
+    
+    company_name = st.selectbox("Select Company", company_names)
+    company_id = get_company_id_by_name(company_name)
+    
+    if company_id:
+        positions = get_all_positions()
+        position_names = ["Add a new position"] + [position['name'] for position in positions]
+        selected_position = st.selectbox("Select Position", position_names)
+        
+        if selected_position == "Add a new position":
+            position_name = st.text_input("Enter Position Name:")
+            remote = st.checkbox("Was the position remote?")
+        else:
+            position_name = selected_position
+            remote = False
+        
+        author_name = "Ally Descoteaux"
+        title = st.text_input("Review Title:")
+        num_co_op = st.number_input("Number of Co-ops:", min_value=0)
+        rating = st.slider("Rating:", min_value=1, max_value=5)
+        recommend = st.checkbox("Recommend")
+        pay_type = st.selectbox("Pay Type:", ["hourly", "salary"])
+        pay = st.number_input("Pay:", min_value=0.0, format="%.2f")
+        job_type = st.text_input("Job Type:")
```

Comment:
For Job Type, i think it would make more sense to have users select between Internship, Co-op, Full-time, Part Time, and Other. Instead of just a text field?